libr/anal/dwarf_process.c

```diff
@@ -0,0 +1,1376 @@
+ * 
+ * @param anal 
+ * @param all_dies Pointer to all DIEs
+ * @param count of all DIEs
+ * @param idx index of the current entry
+ */
+static void parse_typedef(DwContext *ctx, ut64 idx) {
+	const RBinDwarfDie *die = &ctx->all_dies[idx];
+
+	char *name = NULL;
+	char *type = NULL;
+	ut64 size = 0;
+	RStrBuf strbuf;
+	r_strbuf_init (&strbuf);
+	size_t i;
+
+	for (i = 0; i < die->count; i++) {
+		RBinDwarfAttrValue *value = &die->attr_values[i];
+		switch (die->attr_values[i].attr_name) {
+		case DW_AT_name:
+			name = get_die_name (die);
+			if (!name) {
+				goto cleanup;
+			}
+			break;
+		case DW_AT_type:
+			parse_type (ctx, value->reference, &strbuf, &size);
+			type = r_strbuf_drain_nofree (&strbuf);
+			if (!type) {
+				goto cleanup;
+			}
+			break;
+		default:
+			break;
+		}
+	}
+	if (!name) { // type has to have a name for now
+		goto cleanup;
+	}
+	RAnalBaseType *base_type = r_anal_new_base_type (R_ANAL_BASE_TYPE_KIND_TYPEDEF);
+	if (!base_type) {
+		goto cleanup;
+	}
+	base_type->name = name;
+	base_type->type = type;
+	r_anal_save_base_type (ctx->anal, base_type);
+	r_anal_free_base_type (base_type);
+	r_strbuf_fini (&strbuf);
+	return;
+cleanup:
+	free (name);
+	free (type);
+	r_strbuf_fini (&strbuf);
+}
+
+static void parse_atomic_type(DwContext *ctx, ut64 idx) {
+	const RBinDwarfDie *die = &ctx->all_dies[idx];
+
+	char *name = NULL;
+	ut64 size = 0;
+	size_t i;
+	// TODO support endiannity and encoding in future?
+	for (i = 0; i < die->count; i++) {
+		RBinDwarfAttrValue *value = &die->attr_values[i];
+		switch (die->attr_values[i].attr_name) {
+		case DW_AT_name:
+			if (!value->string.content) {
+				name = create_type_name_from_offset (die->offset);
+			} else {
+				name = strdup (value->string.content);
+			}
+			if (!name) {
+				return;
+			}
+			break;
+		case DW_AT_byte_size:
+			size = value->uconstant * CHAR_BIT;
+			break;
+		case DW_AT_bit_size:
+			size = value->uconstant;
+			break;
+		case DW_AT_encoding:
+		default:
+			break;
+		}
+	}
+	if (!name) { // type has to have a name for now
+		return;
+	}
+	RAnalBaseType *base_type = r_anal_new_base_type (R_ANAL_BASE_TYPE_KIND_ATOMIC);
+	if (!base_type) {
+		return;
+	}
+	base_type->name = name;
+	base_type->size = size;
+	r_anal_save_base_type (ctx->anal, base_type);
+	r_anal_free_base_type (base_type);
+}
+
+static const char *get_specification_die_name(const RBinDwarfDie *die) {
+	st32 linkage_name_attr_idx = find_attr_idx (die, DW_AT_linkage_name);
+	if (linkage_name_attr_idx != -1 && die->attr_values[linkage_name_attr_idx].string.content) {
+		return die->attr_values[linkage_name_attr_idx].string.content;
+	}
+	st32 name_attr_idx = find_attr_idx (die, DW_AT_name);
+	if (name_attr_idx != -1 && die->attr_values[name_attr_idx].string.content) {
+		return die->attr_values[name_attr_idx].string.content;
+	}
+	return NULL;
+}
+
+/**
+ * @brief Saves the return type from specification DIE
+ * 
+ * @param all_dies 
```

Comment:
Imho in case we want documentation . Only public apis should be documented. Where and when the discussion about adding such commwnts happened? Are we all suposed to add such java comments on every function now?

Reply:
I addressed this in comment below, but this is just generated basic stuff, it's useless, I am still reworking the parts often so I didn't bother to properly comment it yet, then if anything won't be obvious from the name, arguments or first sight I'll remove such comments